routes/post.go

```diff
@@ -1599,17 +1600,601 @@ func (fes *APIServer) GetDiamondedPosts(ww http.ResponseWriter, req *http.Reques
+				requestData.PostHashHex, err))
+			return
+		}
+		postHash = &lib.BlockHash{}
+		copy(postHash[:], postHashBytes)
+	}
+
+	// Decode the reader public key into bytes. Default to nil if no pub key is passed in.
+	var readerPublicKeyBytes []byte
+	if requestData.ReaderPublicKeyBase58Check != "" {
+		var err error
+		readerPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.ReaderPublicKeyBase58Check)
+		if requestData.ReaderPublicKeyBase58Check != "" && err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Problem decoding user public key: %v : %s", err,
+				requestData.ReaderPublicKeyBase58Check))
+			return
+		}
+	}
+
+	// Get a view with all the mempool transactions.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error constucting utxoView: %v", err))
+		return
+	}
+
+	// Fetch the likers for the post requested.
+	reclouterPubKeys, err := utxoView.GetLikesForPostHash(postHash)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error getting likers %v", err))
+		return
+	}
+
+	// Filter out any restricted profiles.
+	addReaderPublicKey := false
+	pkMapToFilter := make(map[lib.PkMapKey][]byte)
+	for _, pubKey := range reclouterPubKeys {
+		pkMapKey := lib.MakePkMapKey(pubKey)
+		pkMapToFilter[pkMapKey] = pubKey
+		if reflect.DeepEqual(pubKey, readerPublicKeyBytes) {
+			addReaderPublicKey = true
+		}
+	}
+	var filteredPkMap map[lib.PkMapKey][]byte
+	if addReaderPublicKey {
+		filteredPkMap, err = fes.FilterOutRestrictedPubKeysFromMap(pkMapToFilter, readerPublicKeyBytes, "leaderboard" /*moderationType*/)
+	} else {
+		filteredPkMap, err = fes.FilterOutRestrictedPubKeysFromMap(pkMapToFilter, nil, "leaderboard" /*moderationType*/)
+	}
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error filtering out restricted profiles: %v", err))
+		return
+	}
+
+	// Grab verified username map pointer for constructing profile entry responses.
+	verifiedMap, err := fes.GetVerifiedUsernameToPKIDMap()
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("GetLikesForPost: Error fetching verifiedMap: %v", err))
+		return
+	}
+
+	// Create a list of the likers that were not restricted.
+	likers := []*ProfileEntryResponse{}
+	for _, filteredPubKey := range filteredPkMap {
+		profileEntry := utxoView.GetProfileEntryForPublicKey(filteredPubKey)
+		if profileEntry == nil {
+			continue
+		} else {
+			profileEntryResponse := _profileEntryToResponse(profileEntry, fes.Params, verifiedMap, utxoView)
+			likers = append(likers, profileEntryResponse)
+		}
+	}
+
+	// Almost done. Just need to sort the likers.
+	sort.Slice(likers, func(ii, jj int) bool {
+
+		// Attempt to sort on bitclout locked.
+		iiBitCloutLocked := likers[ii].CoinEntry.BitCloutLockedNanos
+		jjBitCloutLocked := likers[jj].CoinEntry.BitCloutLockedNanos
```

Comment:
How are we handling likes from anonymous users?  this looks like it would raise a nil pointer.  okay actually I see that we skip anonymous public key above -- is this the expected behavior? I think it's fine.